packages/compiler-cli/ngcc/src/locking/lock_file_with_child_process/index.ts

```diff
@@ -82,14 +82,15 @@ export class LockFileWithChildProcess implements LockFile {
     this.logger.debug('Forking unlocker child-process');
     const logLevel =
         this.logger.level !== undefined ? this.logger.level.toString() : LogLevel.info.toString();
-
-    const unlocker = fork(this.fs.resolve(__dirname, './unlocker.js'), [path, logLevel], {
-                       detached: true,
-                       stdio: 'pipe',
-                     }) as ChildProcessByStdio<Writable, Readable, Readable>;
-    unlocker.stdout.on('data', data => process.stdout.write(data));
-    unlocker.stderr.on('data', data => process.stderr.write(data));
-
+    const isWindows = process.platform === 'win32';
+    const unlocker = fork(
+                         this.fs.resolve(__dirname, './unlocker.js'), [path, logLevel],
+                         {detached: true, stdio: isWindows ? 'pipe' : 'inherit'}) as
+        ChildProcessByStdio<Writable, Readable, Readable>;
+    if (isWindows) {
+      unlocker.stdout.on('data', process.stdout.write.bind(process.stdout));
+      unlocker.stderr.on('data', process.stderr.write.bind(process.stderr));
+    }
```

Comment:
The `ChildProcessByStdio<Writable, Readable, Readable>` type is not correct any more. (It only has stderr/stdout on Windows.)

You could check for `stderr`/`stdout` explicitly:

```suggestion
    const unlocker = fork(
                         this.fs.resolve(__dirname, './unlocker.js'), [path, logLevel],
                         {detached: true, stdio: isWindows ? 'pipe' : 'inherit'});
    if (unlocker.stdout !== null && unlocker.stderr !== null) {
      unlocker.stdout.on('data', process.stdout.write.bind(process.stdout));
      unlocker.stderr.on('data', process.stderr.write.bind(process.stderr));
    }
```

Or even:

```ts
unlocker.stdout?.on('data', process.stdout.write.bind(process.stdout));
unlocker.stderr?.on('data', process.stderr.write.bind(process.stderr));
```

Reply:
Really like the second option! Thanks.